src/neptune/api/searching_entries.py

```diff
@@ -55,6 +55,26 @@
 SUPPORTED_ATTRIBUTE_TYPES = {item.value for item in AttributeType}
 
 
+class NoLimit(int):
+    def __gt__(self, other: Any) -> bool:
+        return True
+
+    def __lt__(self, other: Any) -> bool:
+        return False
+
+    def __ge__(self, other: Any) -> bool:
+        return True
+
+    def __le__(self, other: Any) -> bool:
+        return False
+
+    def __eq__(self, other: Any) -> bool:
+        return False
+
+    def __ne__(self, other: Any) -> bool:
+        return True
```

Comment:
fyi, `float('inf')` has the same interface, but is of 'float' type so might not be applicable in out case :(

Reply:
I was really conflicted between this and `float("inf")` but mypy would complain because `limit` is `int | None`.
We can disable this particular check with `type: ignore` comment, so what I did here is a proposition that I'm open to discuss :smile:

Reply:
I think you made the right call 😎.
We could try to simplify implementation with [functools.total_ordering](https://docs.python.org/3/library/functools.html#functools.total_ordering) but it doesn't override with superclass method overriding and we want that `int` superclass 😤